keanu-python/tests/test_net.py

```diff
@@ -99,4 +101,17 @@ def test_can_save_and_load(tmpdir) -> None:
     new_net_from_proto = protobuf_loader.load(PROTO_FILE)
     check_loaded_net(new_net_from_proto)
     new_net_from_json = json_loader.load(JSON_FILE)
-    check_loaded_net(new_net_from_json)
\ No newline at end of file
+    check_loaded_net(new_net_from_json)
+
+
+def test_get_vertex_by_label() -> None:
+    vertex = Gamma(1., 1., label="gamma")
+    net = BayesNet([vertex])
+    retrieved_vertex = cast(Vertex, net.get_vertex_by_label("gamma"))
```

Comment:
Is this cast necessary? Seems to work for me without it.

Reply:
This is a mypy cast, so it's required for the build to pass

Reply:
mypy is actually very nifty with how it handles `Optional` - you can do this, with no need for the cast:

    retrieved_vertex = net.get_vertex_by_label("gamma")
    assert retrieved_vertex is not None
    assert retrieved_vertex.get_id() == vertex.get_id()